src/dpp/message.cpp

```diff
@@ -345,6 +353,15 @@ void to_json(json& j, const component& cp) {
 		if (cp.max_values >= 0) {
 			j["max_values"] = cp.max_values;
 		}
+		if (!cp.default_values.empty()) {
+			j["default_values"] = json::array();
+			for (auto const &v : cp.default_values) {
+				json o;
+				o["id"] = v.id;
+				o["type"] = v.type;
```

Comment:
This isn't right, discord doesn't take an int for the type but rather a string. Change `o["type"] = v.type;` to be something like this:

```c++
if(v.type == dpp::cdt_role) {
	o["type"] = "role";
} else if(v.type == dpp::cdt_channel) {
	o["type"] = "channel";
} else if(v.type == dpp::cdt_user) {
	o["type"] = "user";
}
```